packages/mgt-components/src/components/mgt-file/mgt-file.ts

```diff
@@ -0,0 +1,657 @@
+      (this.userId && (!this.itemId && !this.itemPath))
+    ) {
+      driveItem = null;
+    }
+
+    if (this.fileDetails) {
+      driveItem = this.fileDetails;
+    }
+
+    // evaluate to true when only item-id or item-path is provided
+    const getFromMyDrive = !this.driveId && !this.siteId && !this.groupId && !this.listId && !this.userId;
+
+    // todo: error handle can't find file
+
+    if (this.fileQuery) {
+      driveItem = await getDriveItemByQuery(graph, this.fileQuery);
+    } else if (this.itemId && getFromMyDrive) {
+      driveItem = await getMyDriveItemById(graph, this.itemId);
+    } else if (this.itemPath && getFromMyDrive) {
+      driveItem = await getMyDriveItemByPath(graph, this.itemPath);
+    } else if (this.userId) {
+      if (this.itemId) {
+        driveItem = await getUserDriveItemById(graph, this.userId, this.itemId);
+      } else if (this.itemPath) {
+        driveItem = await getUserDriveItemByPath(graph, this.userId, this.itemPath);
+      }
+    } else if (this.driveId) {
+      if (this.itemId) {
+        driveItem = await getDriveItemById(graph, this.driveId, this.itemId);
+      } else if (this.itemPath) {
+        driveItem = await getDriveItemByPath(graph, this.driveId, this.itemPath);
+      }
+    } else if (this.siteId && !this.listId) {
+      if (this.itemId) {
+        driveItem = await getSiteDriveItemById(graph, this.siteId, this.itemId);
+      } else if (this.itemPath) {
+        driveItem = await getSiteDriveItemByPath(graph, this.siteId, this.itemPath);
+      }
+    } else if (this.listId) {
+      driveItem = await getListDriveItemById(graph, this.siteId, this.listId, this.itemId);
+    } else if (this.groupId) {
+      if (this.itemId) {
+        driveItem = await getGroupDriveItemById(graph, this.groupId, this.itemId);
+      } else if (this.itemPath) {
+        driveItem = await getGroupDriveItemByPath(graph, this.groupId, this.itemPath);
+      }
+    } else if (this.insightType) {
+      if (this.userId) {
+        driveItem = await getUserInsightsDriveItemById(graph, this.userId, this.insightType, this.insightId);
+      } else {
+        driveItem = await getMyInsightsDriveItemById(graph, this.insightType, this.insightId);
+      }
+    }
+
+    this.driveItem = driveItem;
+  }
+
+  private getFileTypeIcon() {
+    return;
+    // todo: graph call to get file type
+    // determine which icon to render based on file type
+  }
+
+  private getTextFromProperty(driveItem: MicrosoftGraph.DriveItem, properties: string) {
+    if (!properties || properties.length === 0) {
+      return null;
+    }
+
+    const propertyList = properties.trim().split(',');
+    let text;
+    let i = 0;
+
+    // convert date time
+    const lastModifiedDateTime = new Date(driveItem.lastModifiedDateTime);
+    const relativeDateString = getRelativeDisplayDate(lastModifiedDateTime);
+
+    // convert size to mb
+    const sizeInMb = (driveItem.size / (1024 * 1024)).toFixed(2);
```

Comment:
You can move this in their respective switch case since they are only needed in those cases